jetty-http2/http2-client/src/main/java/org/eclipse/jetty/http2/client/HTTP2ClientConnectionFactory.java

```diff
@@ -58,23 +59,25 @@ public Connection newConnection(EndPoint endPoint, Map<String, Object> context)
 
         Generator generator = new Generator(byteBufferPool, client.getMaxDynamicTableSize(), client.getMaxHeaderBlockFragment());
         FlowControlStrategy flowControl = client.getFlowControlStrategyFactory().newFlowControlStrategy();
-        HTTP2ClientSession session = new HTTP2ClientSession(scheduler, endPoint, generator, listener, flowControl);
+
+        Parser parser = new Parser(byteBufferPool, 4096, 8192);
+        parser.setMaxFrameLength(client.getMaxFrameLength());
+        parser.setMaxSettingsKeys(client.getMaxSettingsKeys());
+
+        HTTP2ClientSession session = new HTTP2ClientSession(scheduler, endPoint, parser, generator, listener, flowControl);
         session.setMaxRemoteStreams(client.getMaxConcurrentPushedStreams());
         long streamIdleTimeout = client.getStreamIdleTimeout();
         if (streamIdleTimeout > 0)
             session.setStreamIdleTimeout(streamIdleTimeout);
 
-        Parser parser = new Parser(byteBufferPool, session, 4096, 8192);
-        parser.setMaxFrameLength(client.getMaxFrameLength());
-        parser.setMaxSettingsKeys(client.getMaxSettingsKeys());
-
         RetainableByteBufferPool retainableByteBufferPool = byteBufferPool.asRetainableByteBufferPool();
-
         HTTP2ClientConnection connection = new HTTP2ClientConnection(client, retainableByteBufferPool, executor, endPoint,
-            parser, session, client.getInputBufferSize(), promise, listener);
+            session, client.getInputBufferSize(), promise, listener);
```

Comment:
Can you rename variable to
```suggestion
            session, client.getInputBufferSize(), sessionPromise, listener);
```